src/targets/maven.ts

```diff
@@ -258,12 +261,92 @@ export class MavenTarget extends BaseTarget {
    * @param distDir directory of the distribution.
    */
   private async uploadDistribution(distDir: string): Promise<void> {
+    const bomFile = await this.getBomFileFomDist(distDir);
+    if (bomFile) {
+      this.logger.debug('Found BOM: ', bomFile);
+      await this.uploadBomDistribution(bomFile);
+    } else {
+      this.logger.debug('Did not find a BOM.');
+      await this.uploadPomDistribution(distDir);
+    }
+  }
+
+  private async getBomFileFomDist(
+    distDir: string
+  ): Promise<string | undefined> {
+    const pomFilepath = join(distDir, POM_DEFAULT_FILENAME);
+    if (await this.isBomFile(pomFilepath)) {
+      return pomFilepath;
+    }
+
+    // There may be several files in the ZIP-ed artifact with the same name,
+    // where the BOM may be one of them (there may not be a BOM). Files may be
+    // renamed when extracting the ZIP, so the default name (`pom-default.xml`)
+    // may not match. It's assumed that any renaming keeps the same extension,
+    // so all files with the same extension are checked to identify the BOM.
+    const filesInDir = await fsPromises.readdir(distDir);
+    const potentialPoms = filesInDir
+      .filter(f => extname(f) === POM_FILE_EXTNAME)
+      .filter(f => f !== POM_DEFAULT_FILENAME)
+      .map(f => join(distDir, f));
```

Comment:
If we wanna be fancy we can do this in a single `reduce` pass, but I like the explicit filter + map.

Reply:
Yep, IMO it's easier to read.